app/auth/apikey_crud.py

```diff
@@ -218,6 +218,11 @@ def check_key(self, api_key: str) -> dict | None:
                 return None
 
             response = row._asdict()
+
+            # If the apikey is revoked
+            if not response["is_active"]:
```

Comment:
Shouldn't an "is_revoked" attribute be added to the DB? 
Is_revoked corresponds to an API Key that has been invalidated manually or by expiry of its date.
is_active corresponds to the user's status in Keycloak

This reactivates an api key when the user's status changes in Keyclaok and is_revoked == False, but not when is_revoked == True.